### PR TITLE
Feature: vc-jwt context

### PR DESCRIPTION
**Pull request by mprorock**
I do not believe that this context is correct (as in where we want to end up), but i believe that it is a good start for discussion

***
<a href="https://pr-preview.s3.amazonaws.com/mesur-io/vc-jwt/pull/54.html" title="Last updated on Feb 16, 2023, 6:55 PM UTC (83a4fde)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jwt/54/aa22ec2...mesur-io:83a4fde.html" title="Last updated on Feb 16, 2023, 6:55 PM UTC (83a4fde)">Diff</a>

### CODE REVIEW

**Review by mkhraisha — APPROVED** [review]
This is a clear path forward on the `@context` issue, allowing you to interoperate cleanly. 

Might need some language around how to process if `@context` is present in the jwt

**Comment by mprorock** [comment]
> Might need some language around how to process if `@context` is present in the jwt

agreed - we could add an issue and reference from this PR, or add to this - happy to take suggestions

the top case i see there is for when multiple contexts are used and not suggested by a `cty` or similar

**Comment by mkhraisha** [comment]
What happens if a context is used that breaks the implied context, for example, if I define a context that makes `iss` only ever to be a string? 
I'm assuming the answer is you use the embedded context over the implied one but we should be clear.

File: examples/vc-2.0/vc-jwt.jsonld [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "https://www.w3.org/2018/credentials/v1",
+    {
+      "vc": "http://www.w3.org/2018/credentials#",
+      "jwt": "http://example.com/jwt#"
+    }
+  ],
+  "@type": "JwtVerifiableCredential",
+  "vc:credentialSubject": {
+    "@type": "JwtCredentialSubject",
+    "jwt:iss": {"@id": "vc:issuer"},
```
Comment (OR13) [inline examples/vc-2.0/vc-jwt.jsonld:12]: this stuff needs to be in the `@context`, but I think the intention is clear.

Comment (mprorock) [reply examples/vc-2.0/vc-jwt.jsonld]: made some adjustments - still working through a double check on n-quads bc json-ld is hard

**Review by decentralgabe — APPROVED** [review]
👏 great demonstration of the concept

**Comment by mprorock** [comment]
i believe there is an implied update in the core data model "Verifiable credentials define terms in a JSON-LD context at `https://www.w3.org/ns/credentials/v2`. Implementers SHOULD include the verifiable credential context in their object definitions. Implementers MAY include additional context as appropriate."

File: index.html [hunk 2]
```diff
@@ -659,6 +659,18 @@ <h2>Version 2</h2>
       <section>
         <h2>Credential Metadata</h2>
         <p><a data-cite="rfc7519#section-5.1">typ</a> MUST use the content type <code>verifiable-credential+jwt</code>.</p>
+        <p>
+          The content type <code>vc+jwt</code> implies the use of an `@context` that defines the registered claims present 
+          in a JWT to provide semantic interoperabilite when using a `vc-jwt` in an RDF capacity.
```
Comment (bumblefudge) [inline index.html:664]:
```suggestion
          in a JWT to provide semantic interoperability when using a `vc-jwt` in an RDF capacity.  
```

Or possibly "a `vc-jwt` that can be deterministically mapped to RDF" ?

Comment (mprorock) [reply index.html:664]: i like that deterministic langugage

Comment (mprorock) [reply index.html:664]: @bumblefudge would `in a JWT to provide semantic interoperability when using a `vc-jwt` and to provide a deterministic method to convert a JWT into RDF using the base [[vc-data-model]] context.` work?

Comment (bumblefudge) [reply index.html:664]:
```suggestion
          in a JWT that provides semantic interoperability and deterministic conversion to RDF via the [[vc-data-model]] and its base context.```
          😎

**Review by selfissued — CHANGES_REQUESTED** [review]
I understand the intent of this PR and it may not be harmful but it also may not be helpful.  It isn't necessary to apply a mapping to VC-JWTs to be able to successfully define and use them.  Adding this to the spec might confuse some people into thinking that it is necessary or recommended.

**Comment by bumblefudge** [comment]
I'm not sure I understand the risk of confusion, is this about the word "implies"?  Would it be more precise to mirror the language of the resolution [recorded here](https://github.com/w3c/vc-data-model/issues/947#issuecomment-1433759132) and word it as a requirement for "bidirectional transformation" rather than an... *implication*? Happy to straw-man a more explicit (although verbose) version if anyone thinks it might be less confusing.

**Comment by bumblefudge** [comment]
I.e.

```
If bidirectional transformation to RDF-based proof types is desired, implementers MUST use the content type <code>vc+jwt</code> and each verifiable credential MUST include an `@context` property set to the appropriate data model version's base context.  Doing so provides semantic interoperability and deterministic conversion to RDF via the [[vc-data-model]] and its base context without compromising the integrity guarantees of either expression or requiring re-issuance or re-signing.
```
happy to cease and desist if I'm misinterpreting what helpful looks like (having not been at the F2F), or continue bikeshedding with further input.

**Comment by TallTed** [comment]
@bumblefudge -- It would be best to codefence your large block in https://github.com/w3c/vc-jwt/pull/54#issuecomment-1434508739 with triple backticks, which do not get interrupted by the single-backticks wrapping the `@context` (though the *do* get displayed, so maybe you want to remove the inner single-backticks)...
~~~
```
blah `blah` blah
```
~~~

**Comment by mprorock** [comment]
@bumblefudge that topic is a whole nother can of worms.  Intent on this PR is to get a starting point to demonstrate how to link a JWT using only registered and private claims back into the graph model and semantics afforded us by the core data model context.  It could for sure, serve as a building block in helping towards the direction indicated by the wg

**Comment by OR13** [comment]
We are not required to add this PR here, just showing that it is possible is enough. 

A few options:

1. A link to this in the appendix
2. An informative comment in the appendix

I agree it could be confusing if not introduced carefully.

**Comment by mprorock** [comment]
Closing in favor of #56

**Pull request closed without merging**